src/trigger.c

```diff
@@ -0,0 +1,292 @@
+                continue;
+            }
+        }
+
+        if (listTrigger->flags.flags & 0x4000) {
+            if (listTrigger->params2 != collisionStatus->unk_0E) {
+                continue;
+            }
+        }
+
+        if (listTrigger->flags.flags & 0x8000) {
+            if (listTrigger->params2 != collisionStatus->unk_10) {
+                continue;
+            }
+        }
+
+        if (listTrigger->flags.flags & 0x100000) {
+            f32* floats;
+            f32 dist;
+
+            if (collisionStatus->bombetteExploded < 0) {
+                continue;
+            }
+
+            floats = listTrigger->params2;
+            dist = dist3D(floats[0], floats[1], floats[2],
+                                collisionStatus->bombetteExplositionPos.x, collisionStatus->bombetteExplositionPos.y,
+                                collisionStatus->bombetteExplositionPos.z);
+
+            if ((floats[3] * 0.5f) + 50.0f < dist) {
+                continue;
+            }
+        }
+
+        if (listTrigger->flags.flags & 0x10000 && get_global_flag(listTrigger->params1) == 0) {
+            continue;
+        }
+
+        if (listTrigger->flags.flags & 0x20000 && get_area_flag(listTrigger->params1) == 0) {
+            continue;
+        }
+
+        listTrigger->flags.flags |= 2;
+    }
+
+    for (i = 0; i < MAX_TRIGGERS; i++) {
+        listTrigger = (*gCurrentTriggerListPtr)[i];
+
+        if (listTrigger == NULL) {
+            continue;
+        }
+
+        if (listTrigger->flags.flags & 1) {
+            if (listTrigger->flags.flags & 2) {
+                if (listTrigger->functionHandler(listTrigger) == 0) {
+                    listTrigger->flags.flags &= ~2;
+                }
+            }
+        }
+    }
+}
+
+void delete_trigger(Trigger* toDelete) {
+    s32 i;
+
+    for (i = 0; i < MAX_TRIGGERS; i++) {
+        if ((*gCurrentTriggerListPtr)[i] == toDelete) {
+            break;
+        }
+    }
+
+    if (i < MAX_TRIGGERS) {
+        heap_free((*gCurrentTriggerListPtr)[i]);
+        (*gCurrentTriggerListPtr)[i] = NULL;
+    }
+}
+
+s32 is_trigger_bound(Trigger* trigger, EvtSource* script) {
+    s32 i;
+
+    for (i = 0; i < MAX_TRIGGERS; i++) {
+        Trigger* listTrigger = (*gCurrentTriggerListPtr)[i];
+
+        if (listTrigger == NULL || listTrigger == trigger) {
+            continue;
+        }
+
+        if (listTrigger->flags.flags & 1) {
+            if (listTrigger->flags.flags & 2) {
+                if (listTrigger->scriptSource == script) {
+                    return TRUE;
+                }
+            }
+        }
+    }
+    return FALSE;
+}
+
+Trigger* get_trigger_by_id(s32 triggerID) {
+    return (*gCurrentTriggerListPtr)[triggerID];
+}
+
+/// @returns TRUE if colliderID is bound to an interaction trigger (press A) and the player can use it.
+s32 should_collider_allow_interact(s32 colliderID) {
+    s32 i;
+
+    if (phys_can_player_interact() == 0) {
```

Comment:
This function logically returns a bool, right? So we can just do
```suggestion
    if (!phys_can_player_interact()) {
```